### DISCUSSION TITLE
scaleBand w/ dates doesn't plot over entire range

### DISCUSSION THREAD

**Discussion by nvolungis** [airbnb/visx#1216] [Q&A] [answered]
I've created a [minimal example here](https://codesandbox.io/s/optimistic-architecture-bgkhc). I have two data points with dates + int values. Dates are plotted against the x and ints against the y. The issue is that y values are cut in half relative to the range. when I use 3 data points, they're cut by 1/3. I'm sure this is a misunderstanding of how to set things up on my part, but I'm pretty stuck. 

Thanks for your help.

**Comment by williaster** [comment]
Hey @nvolungis thanks for checking out `visx`! I get an empty sandbox when I follow that link, could you update it or share another?

**Comment by nvolungis** [comment]
Hey @williaster. I updated the comment with this link here https://codesandbox.io/s/optimistic-architecture-bgkhc. Thanks for looking!

**Comment by williaster** [comment, marked as answer]
Thanks! I converted this to a discussion since it's a usage question not a `visx` issue. I'm not 100% sure I follow what you mean by "y values are cut in half ... cut by 1/3" but I tried to update the sandbox to make it more clear what's happening. what I did:

- added a chart margin, otherwise it was hard to tell where things should be placed
- added a y-axis to more clearly see what values are plotted
- created two lines: a red line which **doesn't** account for scale bandwidth, and a blue one one which **does**. For lines you need to account for scale bandwidth because otherwise the coordinate returned by the band scale maps to the start of the band which is typical for bar chart implementations (where you use `x` + `width`)

![image](https://user-images.githubusercontent.com/4496521/118325932-0de51b00-b4b9-11eb-950b-72dfcc200a2d.png)

As for y-values, I'm not sure if the "cut in half" refers to the fact that you're not including zero in the domain. if you set `zero: true` in your `scaleBand` config it will include zero. Hope that helps!

![image](https://user-images.githubusercontent.com/4496521/118326031-366d1500-b4b9-11eb-9c54-06d371f5a034.png)

here's the [sandbox](https://codesandbox.io/s/naughty-pond-gpb1y?file=/src/App.js).

**Reply by nvolungis** [reply]
This is great thank you so much @williaster! The biggest thing I was missing was that bandwidth value 🙏 .

Regarding the y values... I meant the x values. They just appeared that way because the bandwidth wasn't accounted for. 

Thanks again!
